### PR TITLE
Convert AddressBook framework to C++/WinRT

### PR DESCRIPTION
**Pull request by bviglietta**
This change removes uses of projections from the AddressBook framework and replaces them with C++/WinRT.

### CODE REVIEW

File: build/Tests/Tests.Shared/Tests.Shared.vcxitems [hunk 1]
```diff
@@ -41,6 +41,10 @@
   </ItemGroup>
   <ItemGroup>
     <ClInclude Include="$(MSBuildThisFileDirectory)..\..\..\Frameworks\include\CppUtils.h" />
+    <ClInclude Include="$(MSBuildThisFileDirectory)..\..\..\tests\unittests\Tests.Shared\ByteUtils.h" />
```
Comment (bviglietta) [inline build/Tests/Tests.Shared/Tests.Shared.vcxitems]: ><ClInclude Include="$(MSBuildThisFileDirectory)..\..\..\tests\unittests\Tests.Shared\ByteUtils.h" /> [](start = 4, length = 100)

This change is accidental. I'll remove it.

**Comment by bviglietta** [comment]
@ms-jihua is added to the review. #Closed

File: Frameworks/AddressBook/ABAddressBookManagerInternal.mm [hunk 2]
```diff
@@ -172,22 +200,36 @@ - (id)init {
     return self;
 }
 
+@dynamic contactStore;
```
Comment (DHowett-MSFT) [inline Frameworks/AddressBook/ABAddressBookManagerInternal.mm]: No need for this if you're inmplementing it directly under it. Also, `@dynamic` isn't for manually-implemented properties; it's more for "these will exist at runtime, compiler, I promise"

Comment (bviglietta) [inline Frameworks/AddressBook/ABAddressBookManagerInternal.mm]: I believe I need @dynamic to suppress auto-generation of the field backing the property, don't I?

Comment (DHowett-MSFT) [inline Frameworks/AddressBook/ABAddressBookManagerInternal.mm]: Huh. It looks like [the goal posts keep moving](http://stackoverflow.com/a/9368743) on what/when autosynthesis is engaged. Github does not have an emoticon that is glowering quite enough for me to express myself.

Comment (bviglietta) [inline Frameworks/AddressBook/ABAddressBookManagerInternal.mm]: I would be happy enough to let autosynthesis do its thing, except that in this case I want the property and the synthesized field to have different types.

Comment (DHowett-MSFT) [inline Frameworks/AddressBook/ABAddressBookManagerInternal.mm]: Nah, you're right to use `@dynamic` here. :)

**Pull request merged**